src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/AssessmentService.kt

```diff
@@ -168,7 +172,9 @@ class AssessmentService(
       else -> throw RuntimeException("Assessment type '${assessment::class.qualifiedName}' is not currently supported")
     }
 
-    if (!userAccessService.userCanViewAssessment(user, assessment)) {
+    val isAuthorised = userAccessService.userCanViewAssessment(user, assessment) || (isReferrer && userAccessService.userCanViewApplication(user, assessment.application))
```

Comment:
Is there a reason this logic isn't in the `userCanViewAssessment` function? At the moment the logic is possibly fragmented? If doing it that way, i think `userCanViewAssessment` can directly access the user role (instead of passing `isReferrer` in

Reply:
Thanks for the explanation. I agree `forTimeline` gives context whilst remaining pragmatic 👍